my-app/src/app/main-layout/content-projection/course/course.component.ts

```diff
@@ -1,22 +1,22 @@
 import { DatePipe } from '@angular/common';
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 import { Course } from 'src/app/models/course';
 
 @Component({
   selector: 'app-course',
   templateUrl: './course.component.html',
   styleUrls: ['./course.component.scss']
 })
-export class CourseComponent implements OnInit {
+export class CourseComponent {
 
-  @Input() course: Course;
+  // Task 2
+  @Output() private deleteCourse = new EventEmitter<number>();
 
-  @Output() deleteCourse = new EventEmitter<number>();
+  // Task 2
+  @Input() public course: Course;
 
-  constructor(public datepipe: DatePipe) { }
 
-  ngOnInit(): void {
-  }
+  constructor(public datepipe: DatePipe) { }
 
   public getDate(date: Date): string {
```

Comment:
the fact is you should try to avoid calling methods from template. It may be called on each change detection hook.
So, it may affect application performance.

Reply:
Thanks for the feedback. done!